File: bindings/nodejs/types/secretManager.ts
// Copyright 2021-2022 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

/** Secret manager that uses a Ledger Nano hardware wallet or Speculos simulator. */
export interface LedgerNanoSecretManager {
    /** boolean indicates whether it's a simulator or not. */
    LedgerNano: boolean;
}

/** Secret manager that uses only a mnemonic. */
export interface MnemonicSecretManager {
    Mnemonic: string;
}

/** Secret manager that uses Stronghold. */
export interface StrongholdSecretManager {
    Stronghold: {
        password?: string;
        snapshotPath: string;
    };
}

/** Supported secret managers */
export type SecretManager =
    | LedgerNanoSecretManager
    | MnemonicSecretManager
    | StrongholdSecretManager;

File: bindings/python/native/iota_client/secret_manager.py
class LedgerNanoSecretManager(dict):
    def __init__(self, is_simulator):
        """Initialize a ledger nano secret manager.
        """

        dict.__init__(self, LedgerNano=is_simulator)


class MnemonicSecretManager(dict):
    def __init__(self, mnemonic):
        """Initialize a mnemonic secret manager.
        """

        dict.__init__(self, Mnemonic=mnemonic)


class StrongholdSecretManager(dict):
    def __init__(self, snapshot_path, password):
        """Initialize a stronghold secret manager.
        """

        dict.__init__(self, Stronghold=StrongholdSecretManager.Inner(
            snapshot_path, password))

    class Inner(dict):
        def __init__(self, snapshot_path, password):
            dict.__init__(self, password=password, snapshotPath=snapshot_path)
